### PR TITLE
RBD: build xfsdump and xfsprogs from source; set new expunge set for xfstests

### PR DESCRIPTION
**Pull request by fullerdj**

### CODE REVIEW

**Comment by dachary** [comment]
@fullerdj please ignore the bot, sorry for the inconveniencie

**Comment by jdurgin** [comment]
one thing I didn't mention - each commit should have a Signed-off-by at the end to indicate that the code is contributed under the appropriate license. You can rebase -i and use 'git commit --amend -s' to add it. (git commit -s adds it at commit time).

**Comment by jdurgin** [comment]
This looks good overall. Great to have the reasons catalogued, and verified. Too bad about the hacks to build xfsdump/progs, but we can live with it in this script for now.

File: qa/run_xfstests.sh [hunk 1]
```diff
@@ -46,7 +51,7 @@ TESTS="-g auto"	# The "auto" group is supposed to be "known good"
 
 # rbd presents geometry information that causes mkfs.xfs to
 # issue a warning.  This option avoids this class of problems.
-XFS_MKFS_OPTIONS="-l su=32k"
+#XFS_MKFS_OPTIONS="-l su=32k"
```
Comment (jdurgin) [inline qa/run_xfstests.sh]: ideally moving these options would be in a separate commit, but not a big deal. I'd merged it as is with the signed-off-bys added

Comment (jdurgin) [inline qa/run_xfstests.sh]: it makes me think we might want to add these just for the subset of tests that depend on them in a future update

Comment (fullerdj) [inline qa/run_xfstests.sh]: Actually, this option is obsolete with recent versions of mkfs.xfs (in xfsprogs), which autodetects the correct stripe unit size. I left the comments in case this is ever run (either accidentally or intentionally) with xfsprogs from a distribution. Want me to remove them?

Comment (jdurgin) [inline qa/run_xfstests.sh]: Ah right, I remember you mentioned that now. Yeah, replace them with a comment noting why they're not there
